src/DataGrid.tsx

```diff
@@ -223,10 +227,15 @@ function DataGrid<R, SR, K extends Key>(
   /**
    * defaults
    */
+  const defaultComponents = useDefaultComponents<R, SR>();
   rowHeight ??= 35;
   const headerRowHeight = rawHeaderRowHeight ?? (typeof rowHeight === 'number' ? rowHeight : 35);
   const summaryRowHeight = rawSummaryRowHeight ?? (typeof rowHeight === 'number' ? rowHeight : 35);
-  const RowRenderer = rowRenderer ?? Row;
+  const RowRenderer = components?.rowRenderer ?? defaultComponents?.rowRenderer ?? Row;
+  const sortIcon = components?.sortIcon ?? defaultComponents?.sortIcon ?? SortIcon;
+  const checkboxFormatter =
+    components?.checkboxFormatter ?? defaultComponents?.checkboxFormatter ?? CheckboxFormatter;
+  const noRowsFallback = (components ?? defaultComponents)?.noRowsFallback;
```

Comment:
Actually this wont work if we have for example
```
const components = {};
const defaultComponents = { noRowsFallback: ... };
```
```suggestion
  const noRowsFallback = components?.noRowsFallback ?? defaultComponents?.noRowsFallback;
```